tests/test_asset_list.py

```diff
@@ -289,5 +290,14 @@ def test_write_asset(self):
         self.assertTrue(os.path.exists(os.path.join(target_workdir, test_filename)))
 
 
+def test_write_bytes_asset(tmpdir):
+    asset = get_bytes_asset_dict(
+        b"hello world", target_location="", target_name="hello.txt"
+    )
+    write_asset(asset, tmpdir)
+    with tmpdir.join("hello.txt").open("rb") as f:
+        assert f.read() == b"hello world"
```

Comment:
Can you add a test where the asset is being written to a subdirectory? Copy-pasting this test and making a minor change is sufficient.